pkg/download/download.go

```diff
@@ -75,7 +77,39 @@ func (p *protocol) request(mod, ver string) error {
 }
 
 func (p *protocol) List(ctx context.Context, mod string) ([]string, error) {
-	return p.dp.List(ctx, mod)
+	const op errors.Op = "protocol.List"
+	goList, goErr := p.dp.List(ctx, mod)
+	strList, sErr := p.s.List(ctx, mod)
+	if goErr != nil && sErr != nil {
+		var errs error
+		errs = multierror.Append(errs, goErr, sErr)
+		return nil, errors.E(op, errs)
+	}
+	if goErr != nil && sErr == nil {
+		semver.SortVersions(strList)
+		return strList, nil
+	}
+	if goErr == nil && sErr != nil {
+		return goList, nil
+	}
+
+	combinedList := union(goList, strList)
+	semver.SortVersions(combinedList)
```

Comment:
same, we can remove

Reply:
Your completely right - go sorts the list from proxy anyway.